src/generic.c

```diff
@@ -173,6 +176,7 @@ generic_verify_socket_buffer(int fd, int wndsz)
 	/* Now verify */
 	len = sizeof (wndsz);
 	nwsz = -1;
+	errno = 0;
```

Comment:
Why do you need to set `errno`? One should only look at `errno` once after a failed system call. In other cases, the value is not specified.

Reply:
This is because `generic_verify_socket_buffer()` uses the `ulog_warn()` macros in a case where `errno` is undefined following a successful system call.  In the test output you'd see (depending on your OS implementation) an "`Illegal argument`" string printed following the warning message that a send or receive buffer was not the expected size.

The better fix is to not use the `ulog_warn()` macro and instead use `ulog()` with the `errno` argument explicitly set to zero.  I'll update the patch.

Reply:
Posted an update patch.

Reply:
That makes sense. Please update the patch.

Reply:
Updated.